ray-operator/controllers/ray/utils/util.go

```diff
@@ -71,6 +71,33 @@ func IsCreated(pod *corev1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+// CheckRayHeadRunningAndReady returns (isRayHeadRunning, reason, message) tuple.
+func CheckRayHeadRunningAndReady(ctx context.Context, pods corev1.PodList) (bool, string, string) {
+	log := ctrl.LoggerFrom(ctx)
+	headPodFound := false
+	for _, pod := range pods.Items {
+		if pod.Labels[RayNodeTypeLabelKey] == string(rayv1.HeadNode) {
+			headPodFound = true
+			if pod.Status.Phase != corev1.PodRunning {
+				log.Info(fmt.Sprintf("Head pod is not running; Pod Name: %s; Pod Status.Phase: %v", pod.Name, pod.Status.Phase))
```

Comment:
Avoid using `fmt.Sprintf`. I know there are some places in the codebase that use `fmt.Sprintf` in logging functions. However, it is not a good practice, and we want to remove all such usage from the codebase. See [this doc](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/docs/developer/logging-guidelines.md#message-guidelines) for more details.